3-exceptions/2-after.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+final class PrintFailedException extends \Exception
+{
+}
+
+final class GetReportHandler
+{
+    // ...
+
+    /**
+     * @param int|null $reportId
+     *
+     * @return array
+     * @throws InvalidArgumentException
+     */
+    public function get($reportId)
+    {
+        if ($reportId === null) {
+            throw new InvalidArgumentException("Repord id can never be empty, and needs to have a value");
+        }
+
+        /*
+         * Try-catch only for the scope and not for the full code
```

Comment:
why not rest of the code inside the try block ?

Reply:
That code potentially do not throw a not found exception